torchani/extension/aev.cu

```diff
@@ -0,0 +1,627 @@
+    }
+  }
+  cudaMemcpyToSymbol(csubaev_offsets, subaev_offsets,  sizeof(int) * num_species * num_species);
+  delete[] subaev_offsets;
+}
+
+
+
+//NOTE: assumes size of EtaA_t = Zeta_t = EtaR_t = 1  
+template<typename ScalarRealT = float>
+void
+cuComputeAEV(torch::Tensor coordinates_t,
+	     torch::Tensor species_t,
+	     ScalarRealT Rcr,
+	     ScalarRealT Rca,	     
+	     torch::Tensor EtaR_t,
+	     torch::Tensor ShfR_t,
+	     torch::Tensor EtaA_t,
+	     torch::Tensor Zeta_t,
+	     torch::Tensor ShfA_t,
+	     torch::Tensor ShfZ_t,
+	     torch::Tensor aev_t,
+	     int num_species){
+  const int n_molecules = species_t.size(0);
+  const int max_natoms_per_mol = species_t.size(1);
+  //std::cout << "Running cuComputeAEV with " << n_molecules << "x" << max_natoms_per_mol << "\n";
+
+  AEVScalarParams<float> aev_params;
+  aev_params.Rca = Rca;
+  aev_params.Rcr = Rcr;
+  aev_params.num_species = num_species;
+
+  aev_params.radial_sublength = EtaR_t.size(0) * ShfR_t.size(0); 
+  aev_params.radial_length = aev_params.radial_sublength * num_species;
+
+  aev_params.angular_sublength = EtaA_t.size(0) * Zeta_t.size(0) * ShfA_t.size(0) * ShfZ_t.size(0); 
+  aev_params.angular_length = aev_params.angular_sublength * (num_species * (num_species + 1)/2);
+
+  if(EtaR_t.size(0) != 1 || EtaA_t.size(0) != 1 || Zeta_t.size(0) != 1){
+    std::cerr << "cuda extension is currently not supported for the specified configuration\n";
+    exit(1);
+  }
+  
+  // precompute the aev offsets and load to constand memory
+  initConsts(aev_params);
+
+  // buffer to store all the pairwise distance (Rij)
+  PairDist<float> *d_Rij = 0;
+  auto total_natom_pairs = n_molecules * max_natoms_per_mol * max_natoms_per_mol;
+  CubDebugExit(g_allocator.DeviceAllocate((void**)&d_Rij, sizeof(PairDist<float>) * total_natom_pairs));
+  //cudaMalloc((void **)&d_Rij, sizeof(PairDist<float>) * total_natom_pairs);
+
+  // init all Rij to inf
+  PairDist<float> init;
+  init.Rij = std::numeric_limits<float>::infinity();
+  thrust::fill(thrust::device, d_Rij, d_Rij + total_natom_pairs, init);
+
+  
+  // buffer to store all the pairwise distance that is needed for Radial AEV computation
+  PairDist<float> *d_radialRij = 0;
+  CubDebugExit(g_allocator.DeviceAllocate((void**)&d_radialRij, sizeof(PairDist<float>) * total_natom_pairs));
+  //cudaMalloc((void **)&d_radialRij, sizeof(PairDist<float>) * total_natom_pairs);
+  
+  
+  int *d_count_out = 0;
+  CubDebugExit(g_allocator.DeviceAllocate((void**)&d_count_out, sizeof(int)));
+  //cudaMalloc((void **)&d_count_out, workspace_size);
+  
+  const int block_size = 64;
+  auto start = std::chrono::steady_clock::now();
+  if((species_t.dtype() == torch::kInt32) &&
+     (coordinates_t.dtype() == torch::kFloat32)){
+
+    
+    dim3 block(8, 8, 1);
+    // Compute pairwise distance (Rij) for all atom pairs in a molecule
+    pairwiseDistance 
+      <<<n_molecules, block, sizeof(float) * max_natoms_per_mol * 3>>>

```

Comment:
I think we should run this on stream `c10::cuda::getCurrentCUDAStream()`, instead of the default stream.

Reply:
Working at https://github.com/akkamesh/torchani/pull/1